src/components/pages/teacher-info-page/TeacherInformPage.jsx

```diff
@@ -0,0 +1,197 @@
+import React, { useEffect, useState } from 'react';
+import { getUserData, getClassesData } from '../../../util/DataBaseRequests';
+import { useAuth } from '../../../AuthProvider';
+import { useParams } from 'react-router-dom';
+import Loader from '../../common/Loader';
+
+const TeacherInfoPage = () => {
+  const { userData } = useAuth();
+  const { teacherId } = useParams();
+  const [teacherInfo, setTeacherInfo] = useState(null);
+  const [classes, setClasses] = useState([]);
+  const [classesError, setClassesError] = useState({});
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchTeacherAndClasses = async () => {
+      try {
+        const teacherData = await getUserData(teacherId, userData.token);
+        setTeacherInfo(teacherData.data);
+
+        const response = await getClassesData();
+        const allClasses = response.classes;
+
+        const myClassIds = teacherData.data.myClasses.map((id) => id);
+        const filteredClasses = allClasses.filter((classItem) =>
+          myClassIds.includes(classItem._id)
+        );
+
+        if (filteredClasses.length === 0) {
+          setClassesError({
+            noClassesError: `This teacher hasn't added any classes yet.`,
+          });
+        } else {
+          setClasses(filteredClasses);
+        }
+
+        setIsLoading(false);
+      } catch (error) {
+        error('Error fetching data:', error);
+        setClassesError({
+          fetchError: 'Failed to fetch data. Please try again later.',
+        });
+        setIsLoading(false);
+      }
+    };
+
+    if (userData) {
+      fetchTeacherAndClasses();
+    } else {
+      error('No teacher data found');
+    }
+  }, [teacherId, userData]);
+
+  if (isLoading) return <Loader />;
+  if (classesError.fetchError) return <p>{classesError.fetchError}</p>;
+
+  return (
+    <div className="">
+      <div className="flex flex-col md:flex-row justify-center md:justify-around md:bg-lightGreen">
+        <div className="flex flex-col bg-pureWhite p-4 lg:p-10 ml-4 md:ml-12 md:my-12 rounded md:w-1/2">
+          <div className="flex flex-row gap-4">
+            <div className="flex m-2">
+              {teacherInfo?.profileImageUrl && (
+                <img
+                  src={teacherInfo.profileImageUrl}
+                  alt={`${teacherInfo.firstName} ${teacherInfo.lastName}`}
+                  className="w-20 h-20 rounded-full"
+                />
+              )}
+            </div>
+            <div className="flex justify-center items-center">
+              <h1 className="text-center font-roboto font-medium">
+                {teacherInfo?.firstName} {teacherInfo?.lastName}
+              </h1>
+            </div>
+          </div>
+
```

Comment:
Could you please add here information about teachers education, specialties and Experience

Reply:
added